indica/readers/abstractreader.py

```diff
@@ -185,6 +185,54 @@ def _get_thomson_scattering(
             "method.".format(self.__class__.__name__)
         )
 
+    def get_ppts(
+        self,
+        uid: str,
+        instrument: str,
+        revision: RevisionLike,
+        quantities: Set[str],
+    ) -> Dict[str, Any]:
+
+        database_results = self._get_ppts(uid, instrument, revision, quantities)
+        database_results["channel"] = np.arange(database_results["length"])
+
+        data = {}
+        for quantity in quantities:
+            if "_R" in quantity:
+                dims = ["t", "R_midplane"]
+            elif "_rho" in quantity:
+                dims = ["t", "rho_poloidal"]
+            elif "_data" in quantity:
+                dims = ["t", "channel"]
+            elif "rshift" in quantity:
+                dims = ["t"]
+            else:
+                raise ValueError(f"Unknown quantity: {quantity}")
+
+            data[quantity] = self.assign_dataarray(
```

Comment:
For the data profiles (ne_data, te_data, ...) you should add the additional dimensions rho_data, R_data and z_data to channel.

You're reading them in the st40_reader, but not assigning them here :-).

Reply:
I've just added them as attributes, not sure if there is a better way to handle it with xarray.